cleo/io/inputs/token_parser.py

```diff
@@ -31,7 +31,7 @@ def _parse(self) -> list[str]:
         tokens = []
 
         while self._is_valid():
-            if self._current.isspace():
+            if self._current is not None and self._current.isspace():
```

Comment:
this leads to less changes
```suggestion
            assert self._current is not None # redundant check for mypy
            if self._current.isspace():
```

or even just in-lining the while condition

```python
while self._current is not None:
    ...

```

Reply:
True. I replaced checks and removed the `_is_valid()` method.